### Popularity based Recommender 

- Rely on history of purchase, previously liked, clicked etc 
- Based on whats most popular and are thus not personalized for particular user 
- Often used in systems where they want to recommend what is popular generally such as new sites etc. 


In [23]:
import numpy as np 
import pandas as pd

In [3]:
ratings = pd.read_csv('rating_final.csv')
cusine = pd.read_csv('chefmozcuisine.csv')

In [4]:
ratings.head()

userID  placeID  rating  food_rating  service_rating
0  U1077   135085       2            2               2
1  U1077   135038       2            2               1
2  U1077   132825       2            2               2
3  U1077   135060       1            2               2
4  U1068   135104       1            1               2

In [5]:
cusine.head()

placeID        Rcuisine
0   135110         Spanish
1   135109         Italian
2   135107  Latin_American
3   135106         Mexican
4   135105       Fast_Food

In [13]:
ratings_count = pd.DataFrame(ratings.groupby('placeID').count())
ratings_count.sort_values('rating',ascending=False).head(5)

userID  rating  food_rating  service_rating
placeID                                             
135085       36      36           36              36
132825       32      32           32              32
135032       28      28           28              28
135052       25      25           25              25
132834       25      25           25              25

In [36]:
top_5 = ratings_count.sort_values('rating',ascending=False).head(5).index.tolist()
most_rated = pd.DataFrame(top_5,index=np.arange(5),columns=['placeID'])
summary = pd.merge(most_rated,cusine,on='placeID')
summary

placeID         Rcuisine
0   135085        Fast_Food
1   132825          Mexican
2   135032        Cafeteria
3   135032     Contemporary
4   135052              Bar
5   135052  Bar_Pub_Brewery
6   132834          Mexican

In [38]:
cusine['Rcuisine'].describe()

count         916
unique         59
top       Mexican
freq          239
Name: Rcuisine, dtype: object

##### So based on our popularity based model we know that mexican cusine is one of the most rated or go to cusine and place .. 132825 is the most rated and the second most popular for mexican cusine is place 132834, so we can recommend these 2 atleast based on popularity. 